src/Hamiltonians/abstract.jl

```diff
@@ -144,47 +134,46 @@ julia> rayleigh_quotient(mom, v) # momentum expectation value for state vector `
 momentum
 
 """
-    sm, basis = build_sparse_matrix_from_LO(
-        ham, address=starting_address(ham); cutoff, filter=nothing, nnzs, col_hint, sort=false, kwargs...
-    )
+    build_sparse_matrix_from_LO(
+        ham, address=starting_address(ham);
+        cutoff, filter=nothing, nnzs, col_hint, sort=false, kwargs...
+    ) -> sm, basis
+    build_sparse_matrix_from_LO(ham, addresses::AbstractVector; kwargs...)
 
 Create a sparse matrix `sm` of all reachable matrix elements of a linear operator `ham`
-starting from `address`. The vector `basis` contains the addresses of basis
-configurations.
-
-Providing the number `nnzs` of expected calculated matrix elements may improve performance.
-The default estimates for `nnzs` is `dimension(ham)`.
+starting from `address`. Instead of a single address, a vector of `addresses` can be passed.
+The vector `basis` contains the addresses of basis configurations.
 
-Setting a custom value `col_hint` for the estimated number of nonzero 
-off-diagonal matrix elements in each matrix column may improve performance.
-The default value for `col_hint` is `num_offdiagonals(ham, address)`.
+Providing the number `nnzs` of expected calculated matrix elements and `col_hint` for the
+estimated number of nonzero off-diagonal matrix elements in each matrix column may improve
+performance.
 
 Providing an energy cutoff will skip the columns and rows with diagonal elements greater
 than `cutoff`. Alternatively, an arbitrary `filter` function can be used instead. These are
 not enabled by default.
 
-Setting `sort` to `true` will sort the matrix rows and columns. This is useful when the
-order of the columns matters, e.g. when comparing matrices. Any additional keyword arguments
-are passed on to `Base.sortperm`.
+Instead of a single `address`, a vector of `addresses` can be provided. If no `filter` is
+provided, the matrix will be truncated to the subspace spanned by the `addresses`. For
+generating the matrix with all reachable configurations, use `filter = _ -> true`.
```

Comment:
```suggestion
generating the matrix with all reachable configurations, use `filter = Returns(true)`.
```
This is optional, but we could suggest using the new(ish) `Returns` struct in cases like these.